flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala

```diff
@@ -450,6 +450,39 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
       "1111111111111111111111111111111111111111111111111111111111111111")
   }
 
+  @Test
+  def testRegexReplace(): Unit = {
+    testAllApis(
+      "foobar".regexReplace("oo|ar", ""),
+      "'foobar'.regexReplace('oo|ar', '')",
+      "regex_replace('foobar', 'oo|ar', '')",
+      "fb")
+
+    testAllApis(
+      "foobar".regexReplace("oo|ar", "abc"),
+      "'foobar'.regexReplace('oo|ar', 'abc')",
+      "regex_replace('foobar', 'oo|ar', 'abc')",
+      "fabcbabc")
+
+    testAllApis(
+      'f33.regexReplace("oo|ar", ""),
+      "f33.regexReplace('oo|ar', '')",
+      "REGEX_REPLACE(f33, 'oo|ar', '')",
+      "null")
+
+    testAllApis(
+      "foobar".regexReplace('f33, ""),
+      "'foobar'.regexReplace(f33, '')",
+      "REGEX_REPLACE('foobar', f33, '')",
+      "null")
+
+    testAllApis(
+      "foobar".regexReplace("oo|ar", 'f33),
+      "'foobar'.regexReplace('oo|ar', f33)",
+      "REGEX_REPLACE('foobar', 'oo|ar', f33)",
+      "null")
+  }
+
```

Comment:
what about `"fooor".regexReplace("oo", "")`? is it `"for"` or `"fr"`? the problem is what about the complicated/corner case, which maybe worth a test to cover it.